docs/pages/versions/unversioned/introduction/testing-with-jest.md

```diff
@@ -0,0 +1,173 @@
+
+`yarn add react-dom enzyme enzyme-adapter-react-16 --dev` **or** `npm i react-dom enzyme enzyme-adapter-react-16 --save-dev`
+
+With enzyme's adapter for react < 16, we need to have this set up before every test that runs. Jest gives us a way to do that, head back over to `package.json`:
+
+```js
+"jest": {
+  "preset": "jest-expo",
+  "setupFiles": [
+    "<rootDir>/jest.setup.js"
+  ]
+}
+```
+
+Now create the file `jest.setup.js`
+
+```js
+import { configure } from 'enzyme';
+
+import Adapter from 'enzyme-adapter-react-16';
+
+configure({ adapter: new Adapter() });
+```
+
+That's it! Now we can start writing Jest tests!
+
+## My First Jest Test
+
+We are going to write a test for `App.js` by creating `App.test.js`, Jest will identify this as a test, and include it in the tests queue. There are other ways to [structure your tests](#structure-your-tests), but we will cover that later in this tutorial.
+
+`App.test.js`:
+```js
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import App from './App';
+
+const shallowApp = shallow(<App />);
+
+describe('<App />', () => {
+  it('renders correctly', () => {
+    expect(shallowApp).toMatchSnapshot();
+  });
+});
+```
+
+Now run `yarn test` or `npm run test`, if all went well you should see a snapshot created and 1 test passed!
+
+> **Node Version**: If you are running node version 11.11, [Jest will throw errors](https://github.com/facebook/jest/issues/8069), please downgrade or upgrade to 11.12+
+
+This was a very simple test, for more information take a look at the following links:
+
+- [Api: Globals](https://jestjs.io/docs/en/api)
+- [Api: Mock Functions](https://jestjs.io/docs/en/mock-function-api)
+- [Cli Options](https://jestjs.io/docs/en/cli)
+
+## Code Coverage Reports
+
+Running tests are cool and all, but wouldn't you like to see your Expo projects total code coverage?! Maybe even see it in html format?! This section is for you!
+
+Let's head back to `package.json` and add the following:
+
+```js
+"jest": {
+  ...
+  "collectCoverage": true,
+  "collectCoverageFrom": [
+    "**/*.{js,jsx}",
+    "!**/coverage/**",
+    "!**/node_modules/**",
+    "!**/babel.config.js",
+    "!**/jest.setup.js"
+  ],
+  "verbose": true
+}
+```
+
+The above additions let's Jest know to collect coverage of all ***.js & .jsx*** file types and not within ***/coverage***, ***/node_modules/*** and our 2 project config files (add/remove more exclusions to this list to match your Expo app needs). With [verbose](https://jestjs.io/docs/en/cli#verbose) set as true.
+
+Now run the test again, you should see **/coverage/** in your app directory! Find the `index.html` file within and double click to open it up in a browser. Not only do we have reporting in our cli, we also have an html version of our code coverage, pretty cool stuff!
+
+> **Note**: You probably don't want to upload the html reporting to git, so please remember to add `coverage/\*\*/*` as a line in `.gitignore`
+
+## Structure your Tests
```

Comment:
we should explain that the standard way to do it is this but you can do whatever you like

Reply:
this has been updated